pkg/cmd/run.go

```diff
@@ -60,6 +59,14 @@ func RunMicroshift(cfg *config.MicroshiftConfig, flags *pflag.FlagSet) error {
 		klog.Fatalf("Error in reading and validating flags: %v", err)
 	}
 
+	if cfg.Cluster.ServiceCIDR != "" {
+		clusterDNS, err := config.GetClusterDNS(cfg.Cluster.ServiceCIDR)
+		if err != nil {
+			klog.Fatalf("failed to get DNS IP: %v", err)
+		}
+		cfg.Cluster.DNS = clusterDNS
+	}
+
```

Comment:
Since this PR removes "DNS" from the configuration API, I disagree that we want errors generated by the computation of this IP to be treated as validation errors, or for that computation to happen as part of config reading/validation. Looking at the implementation, it seems that currently the only way this would return an error is if the Service CIDR is invalid or has fewer than 10 IPs. Those might be candidates for validation?

Reply:
I'd also expect an error from the DNS Service IP computation if there is no Service CIDR configured instead of ignoring it and pushing the failure later.

Reply:
I don't think we're disagreeing here. My point is that following a `ReadAndValidate()` you either have a valid configuration (whether the parameter is directly configurable by the user like Service CIDR or auto-configured based on user configurable input like Cluster DNS) or you don't. In the latter case, you need to fail fatally.

So, yes, you need to validate that Service CIDR is non-empty and in valid format plus that it's big enough to contain the Cluster IP. Whether you test that in a way that the following `net.ParseCIDR(serviceCIDR)` and `cidr.Host(service, 10)` can never fail or you let them fail and report that failure is probably a matter of taste.